### DISCUSSION TITLE
The number of room connections is negative

### DISCUSSION THREAD

**Discussion by LionSummer** [colyseus/colyseus#609] [Q&A]
Negative numbers appear in the clients field stored in the driver mongodb in the creation service
![image](https://github.com/colyseus/colyseus/assets/38936611/e317db09-7f4a-479a-b8ef-be404dbf56d9)

**Comment by endel** [comment]
HI @LionSummer, most of the time I've seen this behavior was due to missing `await` for `allowReconnection` calls during `onLeave`:

```typescript
async onLeave(client) {
  // ...

  // here
  await this.allowReconnection(client, /*  */);

  // ...
}
```

Can you check how yours is implemented? 

Also, the MongoDB driver is not the recommendation anymore, if you can please use the Redis driver.

**Reply by LionSummer** [reply]
How to use the redis driver ? 
Does version 0.14 support redis driver?

**Reply by LionSummer** [reply]
@endel Hi， Already in the onLeave method Called allowReconnection, also await，
![image](https://github.com/colyseus/colyseus/assets/38936611/ec3da516-fa2f-4f2b-88ee-1dd053f86478)

**Reply by vinkent420** [reply]
I'm also using ver 0.14 and dont know if Colyseus support redis driver ?

**Comment by LionSummer** [comment]
> I'm also using ver 0.14 and dont know if Colyseus support redis driver ?

of course，I tried.

**Reply by LionSummer** [reply]
When the list of rooms becomes larger, searching for rooms will become slower and slower. I think it is because the "getRooms" interface used in the query JSON.parse all the rooms.
@endel  
Is there any way to optimize or solve the above situation?
